src/relayserver/RelayServer.ts

```diff
@@ -478,23 +493,23 @@ export class RelayServer extends EventEmitter {
     spam('logs?', logs)
     spam('options? ', options)
     const decodedLogs = abiDecoder.decodeLogs(logs).map(this._parseEvent)
-    let receipt
+    let receipts: TransactionReceipt[] = []
     // TODO: what about 'penalize' events? should send balance to owner, I assume
     // TODO TODO TODO 'StakeAdded' is not the event you want to cat upon if there was no 'HubAuthorized' event
     for (const dlog of decodedLogs) {
       switch (dlog.name) {
         case 'HubAuthorized':
-          receipt = await this._handleHubAuthorizedEvent(dlog)
+          receipts = receipts.concat(await this._handleHubAuthorizedEvent(dlog))
```

Comment:
Please do not use 'await' in this manner. Please create variables for them, it greatly improves readability (at least for me)